src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -456,6 +712,10 @@ void CodegenLLVMVisitor::visit_codegen_var_list_statement(
 }
 
 void CodegenLLVMVisitor::visit_double(const ast::Double& node) {
+    if (is_kernel_code && vector_width > 1) {
```

Comment:
got bit confused here : In my mind `compute loop body` as well as outside loop code is part of `kernel`. May be we should rename this to `is_loop_body_code` or something similar?

Reply:
This is done to handle the case when we visit a function that is not a kernel/not vector

```
// Not vectorised
PROCEDURE() {
  some constant = 2
}

// Vectorised
VOID nrn_state(...) { .. }
```

Reply:
What we need to think about, is that if we have a function call (e.g.`rates(voltage)` in original HH), how we proceed in this case?
- Inline function?
- Vectorise function (`rates` here?)
etc.

Reply:
This is good point.

> Inline function?

We already do this at NMODL level : if you pass the flag `passes --inline` then all procedures call are inlined inside AST.

> Vectorise function (rates here?)

that's another option but for all current use cases, DSL level inlining will be sufficient. But I think this deserve a ticket!